applications/StructuralMechanicsApplication/custom_constitutive/truss_plasticity_constitutive_law.cpp

```diff
@@ -0,0 +1,294 @@
+//************************************************************************************
+
+double& TrussPlasticityConstitutiveLaw::GetValue(
+    const Variable<double>& rThisVariable,
+    double& rValue
+    )
+{
+    if(rThisVariable == PLASTIC_STRAIN) rValue = this->mAccumulatedPlasticStrain;
+    else if(rThisVariable == PLASTIC_ALPHA) rValue = this->mPlasticAlpha;
+    else KRATOS_ERROR << "Can't get the specified value" << std::endl;
+    return rValue;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+array_1d<double, 3 > & TrussPlasticityConstitutiveLaw::GetValue(
+    const Variable<array_1d<double, 3 > >& rThisVariable,
+    array_1d<double, 3 > & rValue)
+{
+    if (rThisVariable == FORCE)
+    {
+        rValue[0] = this->mStressState;
+        rValue[1] = 0.0;
+        rValue[2] = 0.0;
+    }
+    else KRATOS_ERROR << "Can't get the specified value" << std::endl;
+    return rValue;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void TrussPlasticityConstitutiveLaw::SetValue(
+    const Variable<double>& rThisVariable,
+    const double& rValue,
+    const ProcessInfo& rCurrentProcessInfo
+    )
+{
+    if(rThisVariable == PLASTIC_STRAIN) this->mAccumulatedPlasticStrain = rValue;
+    if(rThisVariable == PLASTIC_ALPHA) this->mPlasticAlpha = rValue;
+    else KRATOS_ERROR << "Can't set the specified value" << std::endl;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void TrussPlasticityConstitutiveLaw::SetValue(const Variable<bool>& rThisVariable,
+                          const bool& rValue,
+                          const ProcessInfo& rCurrentProcessInfo)
+{
+    if(rThisVariable == INELASTIC_FLAG) this->mInelasticFlag = rValue;
+    else KRATOS_ERROR << "Can't set the specified value" << std::endl;
+}
+
+
+//************************************************************************************
+//************************************************************************************
+
+double& TrussPlasticityConstitutiveLaw::CalculateValue(
+    ConstitutiveLaw::Parameters& rParameterValues,
+    const Variable<double>& rThisVariable,
+    double& rValue
+    )
+{
+    if(rThisVariable == TANGENT_MODULUS)
+    {    
+        const Properties& r_material_properties = rParameterValues.GetMaterialProperties(); 
+        const double hardening_modulus = r_material_properties[HARDENING_MODULUS_1D];
+        const double youngs_modulus = r_material_properties[YOUNG_MODULUS];
+        if (this->mInelasticFlag) 
+        {
+            rValue = (hardening_modulus*youngs_modulus)/(hardening_modulus+youngs_modulus);;
```

Comment:
maybe add a debug check for division by zero (=> KRATOS_DEBUG_ERROR_IF...)

Otherwise I could make this fail